private-templates-service/client/src/store/templates/actions.ts

```diff
@@ -23,7 +23,24 @@ export function failGetAll(error: IncomingMessage): AllTemplateAction {
     error: error
   }
 }
-
+export function getTemplatesByTags(tags?: string[]): (dispatch: any, getState: () => RootState) => void {
```

Comment:
do we need this as a separate action? can we modify the existing `getAllTemplate` action to accept an optional param called tags?

Reply:
yes, we can do that